### PR TITLE
Issue 1480 - switched to using goautoneg package for parsing accept headers

### PR DESCRIPTION
**Pull request by soltysh**
This should address the problem described in #1480. 

@liggitt PTAL

### CODE REVIEW

**Comment by soltysh** [comment]
[test]

File: pkg/cmd/server/origin/master.go [hunk 1]
```diff
@@ -14,6 +14,8 @@ import (
 	"strings"
 	"time"
 
+	"bitbucket.org/ww/goautoneg"
```
Comment (liggitt) [inline pkg/cmd/server/origin/master.go]: need to add godeps?

File: pkg/cmd/server/origin/master.go [hunk 2]
```diff
@@ -372,13 +374,14 @@ func initAPIVersionRoute(root *restful.WebService, versions ...string) {
 // and the Accept header supports text/html
 func assetServerRedirect(handler http.Handler, assetPublicURL string) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
-		accept := req.Header.Get("Accept")
-		if req.URL.Path == "/" && strings.Contains(accept, "text/html") {
-			http.Redirect(w, req, assetPublicURL, http.StatusFound)
-		} else {
-			// Dispatch to the next handler
-			handler.ServeHTTP(w, req)
+		accepts := goautoneg.ParseAccept(req.Header.Get("Accept"))
```
Comment (liggitt) [inline pkg/cmd/server/origin/master.go]: don't parse unless req.URL.Path == '/'

File: pkg/cmd/server/origin/master.go (hunk 1 above)
Comment (soltysh) [inline pkg/cmd/server/origin/master.go]: We already have that in place :smile:

File: pkg/cmd/server/origin/master.go [hunk 3]
```diff
@@ -372,13 +374,14 @@ func initAPIVersionRoute(root *restful.WebService, versions ...string) {
...
+		for _, accept := range accepts {
+			if req.URL.Path == "/" && accept.Type == "text" && accept.SubType == "html" {
+				http.Redirect(w, req, assetPublicURL, http.StatusFound)
```
Comment (liggitt) [inline pkg/cmd/server/origin/master.go]: need a `return`

**Comment by soltysh** [comment]
@liggitt comments addressed.

**Comment by liggitt** [comment]
LGTM, [merge]

**Pull request merged**